### PR TITLE
Migrate WelcomeScreen to TS and Reanimated 2

### PR DESCRIPTION
**Pull request by jkadamczyk**
Fixes RNBW-3704

## What changed (plus any additional context for devs)
* Rewrite to TS
* Rewrite to Ranimated 2 - 100%
* Break down some components – cleaner code

## PoW (screenshots / screen recordings)
Before:
iOS: https://streamable.com/708g6q
Android: https://streamable.com/5ixsme

After:
iOS: https://streamable.com/n0ggsm
Android: https://streamable.com/na0tar

## Dev checklist for QA: what to test
Check animations:
- Rainbows springing on entry all over the backround
- Buttons and text UI springing with scale a bit on screen entry
- New wallet button scale pusling
- New wallet button colorful border
- New wallet button colorful shadow on iOS
- rainbow text above buttons colorful mask

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why – animations
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/screens/WelcomeScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+            onPress={showRestoreSheet}
+            shadowStyle={sx.existingWalletShadow}
+            style={[
+              sx.existingWallet,
+              { backgroundColor: colors.blueGreyDarkLight },
+            ]}
+            testID="already-have-wallet-button"
+            text={lang.t('wallet.new.already_have_wallet')}
+            textColor={colors.alpha(colors.blueGreyDark, 0.8)}
+          />
```
Comment (estebanmino) [inline src/screens/WelcomeScreen.tsx:370]: christian made some changes to this screen that are already merged to the .js version of this file as https://github.com/rainbow-me/rainbow/pull/3434/files#diff-5b1569bcf62ccdac475084998a3f2edf9d94ecfaca4d92cd3a48864629a61afcR568-R588 that i don't see here

Comment (jkadamczyk) [reply src/screens/WelcomeScreen.tsx:370]: Working on it

**Review by ibrahimtaveras00 — APPROVED** [review]

**Review by estebanmino — APPROVED** [review]
🚀

**Review by tchayen — APPROVED** [review]
Looks good

**Pull request merged**